src/albert/collections/data_templates.py

```diff
@@ -251,3 +244,38 @@ def delete(self, *, id: str) -> None:
             The ID of the data template to delete.
         """
         self.session.delete(f"{self.base_path}/{id}")
+
+    def get_all(
+        self,
+        *,
+        name: str | None = None,
+        user_id: str | None = None,
+        order_by: OrderBy = OrderBy.DESCENDING,
+        limit: int = 100,
+        offset: int = 0,
+    ) -> Iterator[DataTemplate]:
+        """
+        Retrieve fully hydrated DataTemplate objects with optional filters.
+        Uses `get_by_ids` in batches to hydrate the results for convenience.
+
+        Use :meth:`search` for better performance.
+        """
+
+        def batched(iterable, size):
+            """Yield lists of up to `size` IDs from an iterable of objects with an `id` attribute."""
+
+            it = (item.id for item in iterable)
+            while batch := list(islice(it, size)):
+                yield batch
+
+        id_batches = batched(
+            self.search(name=name, user_id=user_id, order_by=order_by, limit=limit, offset=offset),
+            100,  # batch size
```

Comment:
I think we should set the batch size equal to the `limit` keyword argument (soon to be renamed to `page_size`), such that every time we yield a page from the `search` API call we yield that entire batch as hydrated objects.